### DISCUSSION TITLE
About registration of a TestWatcher extension

### DISCUSSION THREAD

**Discussion by ghost** [junit-team/junit-framework#3993] [General]
I'm planning to start using a newer version of the framework and I saw this warning in the TestWatcher documentation:

> If a TestWatcher is registered via a non-static (instance) field – for example, using @RegisterExtension – and the test class is configured with @ TestInstance(Lifecycle.PER_METHOD) semantics (which is the default lifecycle mode), the TestWatcher will not be invoked with events for @ TestTemplate methods (for example, @RepeatedTest or @ParameterizedTest).

Maybe I got something wrong, but with the JUnit 5.11.0 this code outputs "Failed"

    @RegisterExtension
    TestWatcher testWatcher = new TestWatcher() {

        @Override
        public void testFailed(ExtensionContext context, Throwable cause) {
            System.out.println("Failed");
        }
    };

    @RepeatedTest(1)
    void test() {
        Assertions.assertTrue(true);
    }

    @AfterEach
    void tearDown() {
        Assertions.fail();
    }

The test class configuration is default (Lifecycle.PER_METHOD).

**Comment by marcphilipp** [comment]
This paragraph was added to address #3234. There was no behavioral change. `TestWatcher` extensions registered this way receive events for _invocations_ of `@TestTemplate` methods, but not for the method itself if it's `@Disabled`.
